datafusion/sqllogictest/test_files/scalar.slt

```diff
@@ -1768,52 +1768,61 @@ SELECT make_array(1, 2, 3);
 [1, 2, 3]
 
 # coalesce static empty value
-query T
-SELECT COALESCE('', 'test')
+query TT
+SELECT       COALESCE('', 'test'),
+arrow_typeof(COALESCE('', 'test'))
 ----
-(empty)
+(empty) Utf8
 
 # coalesce static value with null
-query T
-SELECT COALESCE(NULL, 'test')
+query TT
+SELECT       COALESCE(NULL, 'test'),
+arrow_typeof(COALESCE(NULL, 'test'))
 ----
-test
-
+test Utf8
 
+# Create table with a dictionary value
 statement ok
 create table test1 as values (arrow_cast('foo', 'Dictionary(Int32, Utf8)')), (null);
 
-# test coercion string
-query ?
-select  coalesce(column1, 'none_set') from test1;
+# test coercion string (should preserve the dictionary type)
+query ?T
+select       coalesce(column1, 'none_set'),
+arrow_typeof(coalesce(column1, 'none_set'))
+from test1;
 ----
-foo
-none_set
+foo Dictionary(Int32, Utf8)
+none_set Dictionary(Int32, Utf8)
 
-# test coercion Int
-query I
-select coalesce(34, arrow_cast(123, 'Dictionary(Int32, Int8)'));
+# test coercion Int and Dictionary
+query ?T
```

Comment:
As @erratic-pattern  noted the difference here is that now that the output type of cealesce is `Dictionary` rather than Int.